### PR TITLE
docs(aio): Updated router guide content and examples

### PR DESCRIPTION
**Pull request by brandonroberts**
## PR Checklist
Does please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

cc: @wardbell

### CODE REVIEW

File: aio/content/examples/ngmodule/src/app/hero/hero-detail.component.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export class HeroDetailComponent implements OnInit {
     private heroService: HeroService) { }
 
   ngOnInit() {
-    let id = parseInt(this.route.snapshot.params['id'], 10);
+    let id = parseInt(this.route.snapshot.paramMap.get('id'), 10);
     this.heroService.getHero(id).then(hero => this.hero = hero);
```
Comment (wardbell) [inline aio/content/examples/ngmodule/src/app/hero/hero-detail.component.ts]: I've asked the Angular team why we can't use indexing and write `paramMap['id'])`.

**Review by wardbell — APPROVED** [review]
Thanks, Brandon. This is great and an important update. 

My commit makes minor textual and formatting changes. Leave a note here agreeing to them and I think we can merge.

**Comment by brandonroberts** [comment]
@wardbell changes LGTM

**Comment by wardbell** [comment]
squashed and marked for merge

**Pull request merged**